test/integration/scheduler/scheduler_test.go

```diff
@@ -554,18 +579,53 @@ var _ = ginkgo.Describe("Scheduler", func() {
 
 		// Make sure workloads are in the same scheduling cycle.
 		testBEClusterQ := testing.MakeClusterQueue("test-be-cq").
-			Cohort("be").QueueingStrategy(kueue.BestEffortFIFO).
+			Cohort("be").
 			Resource(testing.MakeResource(corev1.ResourceCPU).
 				Flavor(testing.MakeFlavor(onDemandFlavor.Name,
 					"15").Max("15").Obj()).
 				Obj()).
 			Obj()
 		gomega.Expect(k8sClient.Create(ctx, testBEClusterQ)).Should(gomega.Succeed())
 		defer func() {
+			gomega.Expect(framework.DeleteClusterQueue(ctx, k8sClient, prodBEClusterQ)).Should(gomega.Succeed())
+			gomega.Expect(framework.DeleteClusterQueue(ctx, k8sClient, devBEClusterQ)).Should(gomega.Succeed())
 			gomega.Expect(framework.DeleteClusterQueue(ctx, k8sClient, testBEClusterQ)).Should(gomega.Succeed())
 		}()
 
 		framework.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, prodBEClusterQ.Name, wl1)
 		framework.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, devBEClusterQ.Name, wl2)
 	})
+
+	ginkgo.It("Should schedule workloads by their priority strictly in StrictFIFO", func() {
+		strictFIFOClusterQ := testing.MakeClusterQueue("strict-fifo-cq").
+			QueueingStrategy(kueue.StrictFIFO).
+			Resource(testing.MakeResource(corev1.ResourceCPU).
+				Flavor(testing.MakeFlavor(onDemandFlavor.Name,
+					"5").Max("5").Obj()).
+				Obj()).
+			Obj()
+		gomega.Expect(k8sClient.Create(ctx, strictFIFOClusterQ)).Should(gomega.Succeed())
+		defer func() {
+			gomega.Expect(framework.DeleteClusterQueue(ctx, k8sClient, strictFIFOClusterQ)).Should(gomega.Succeed())
+		}()
+
+		strictFIFOQueue := testing.MakeQueue("strict-fifo-q", ns.Name).ClusterQueue(strictFIFOClusterQ.Name).Obj()
+		ginkgo.By("Creating workloads")
+		wl1 := testing.MakeWorkload("wl1", ns.Name).Queue(strictFIFOQueue.
+			Name).Request(corev1.ResourceCPU, "2").Priority(pointer.Int32(100)).Obj()
+		gomega.Expect(k8sClient.Create(ctx, wl1)).Should(gomega.Succeed())
+		wl2 := testing.MakeWorkload("wl2", ns.Name).Queue(strictFIFOQueue.
+			Name).Request(corev1.ResourceCPU, "5").Priority(pointer.Int32(10)).Obj()
+		gomega.Expect(k8sClient.Create(ctx, wl2)).Should(gomega.Succeed())
+		// wl3 can't be scheduled before wl2 even though there is enough quota.
+		wl3 := testing.MakeWorkload("wl3", ns.Name).Queue(strictFIFOQueue.
+			Name).Request(corev1.ResourceCPU, "1").Priority(pointer.Int32(1)).Obj()
+		gomega.Expect(k8sClient.Create(ctx, wl3)).Should(gomega.Succeed())
+
+		gomega.Expect(k8sClient.Create(ctx, strictFIFOQueue)).Should(gomega.Succeed())
+
+		framework.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, strictFIFOClusterQ.Name, wl1)
+		framework.ExpectWorkloadsToBePending(ctx, k8sClient, wl2)
+		framework.ExpectWorkloadsToBeInadmissible(ctx, k8sClient, wl3)
```

Comment:
I think wl3 will never be tried by the scheduler, so its condition in status will never be set. I guess the thing we need to check here is that the admission field is consistently not set.

Reply:
Yes. I change it to consistently check.

By the way, its condition in status is set as `Inadmissible Queue strict-fifo-q doesn't exist`
https://github.com/kubernetes-sigs/kueue/blob/82201f8ca8bc245324800b3a89747d173c55767e/pkg/controller/core/workload_controller.go#L85

Reply:
But the queue does exist, it is just that the workload will never be at the head of the clusterQueue that that queue references and so will never be tried.

did you ever see it marked as Inadmissible in the test?